### PR TITLE
Update OpenFeature SDK

### PR DESCRIPTION
**Pull request by tapico-weyert**
Update the PostHog provider for OpenFeature SDK 

- Updated the OpenFeature JS SDK to the latest
- Updated code to match the JS SDK
- Removed unnecessary code that is now covered by the updated SDK

### CODE REVIEW

File: src/hooks/LoggingHook.ts [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+    console.log(
+      `Running 'finally' logger hook for flag: ${hookContext.flagKey}`
+    );
   }
 
-  error(hookContext: HookContext, error: Error, _hookHints: HookHints) {
-    console.log(`Running 'error' logger hook for flag: ${hookContext.flagKey}`)
-    console.error(error)
+  error(hookContext: HookContext, err: Error) {
+    console.log(`Running 'error' logger hook for flag: ${hookContext.flagKey}`);
```
Comment (fiendfiend) [inline src/hooks/LoggingHook.ts]: trailing semicolons have somehow appeared in this file

File: example/index.ts [hunk 2]
```diff
@@ -33,19 +38,34 @@ async function start() {
...
+  console.log(`\n\nstringResultDetails:`, stringVariant)
+
+  const booleanResultDetails2 = await client.getBooleanDetails('dummy2', true, {
+    targetingKey,
+    groups: { "account-servicer": '123456789' },
+    personalProperties: {
+      "email": "weyert@mailbox.com"
+    },
+    groupProperties: {
+      name: 'Weyert',
```
Comment (fiendfiend) [inline example/index.ts]: are you ok with it having your PII in this?

Comment (weyert) [reply example/index.ts]: Changed it, it's not like my name is not already out in the wild, lol :D

File: src/hooks/LoggingHook.ts (hunk 1 above)
Comment (weyert) [reply src/hooks/LoggingHook.ts]: Reran Prettier should be resolved now

**Pull request merged**